lambda/translmod.mli

```diff
@@ -25,7 +25,7 @@ type compilation_unit_style =
   | Set_individual_fields (* Closure *)
 
 val transl_implementation:
-      Compilation_unit.t -> structure * module_coercion
+      Compilation_unit.t -> structure * module_coercion * module_coercion option
```

Comment:
It should really be `Typedtree.implementation` rather than a tuple appearing out of nowhere, but that turns out to be untenable for _unbelievably_ bad reasons. Suffice it to say that making this API not shitty is beyond the scope of this PR. Adding a comment instead.